### ISSUE TITLE
I lost ~12 hours of messages.

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-web#21052]
### Steps to reproduce

0. I closed my laptop around 01:00
1. I changed my nick on EI around 07:00
2. I opened desktop a while late, around 10:00
3. It tried to incremental sync for *ages* for an hour or so (while on somewhat patchy connectivity)
4. I restarted it at to try to kick it into life at some point
5. Eventually it started syncing again and I can send msgs... but i have ~12 hours of history missing in all(?) of my rooms.

#### What did you expect?

We should never lose history.

#### What happened instead?

We lost history.

### Operating system

macOS

### Application version

Nightly

### How did you install the app?

Nightly

### Homeserver

matrix.org

### Will you send logs?

Yes

**Comment by ShadowJonathan** [comment]
Is this encrypted history? If it's normal history, a clear-cache and refresh should fix this.

Also, this would probably be fixed by a wider upgrade to sliding sync.

**Comment by turt2live** [comment]
> Is this encrypted history? If it's normal history, a clear-cache and refresh should fix this.

No, and a clear cache & reload is not a suitable workaround. We have a ton of code which tries to fill gaps, and there's a bug somewhere in there.
